File: engine/engine-model/src/main/scala/tools/packet/monster/DamageMonsterFriendly.scala
package tools.packet.monster

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class DamageMonsterFriendly(private var _objectId: Int, private var _damage: Int, private var _remainingHp: Int, private var _maximumHp: Int) extends PacketInput {
  def objectId: Int = _objectId

  def damage: Int = _damage

  def remainingHp: Int = _remainingHp

  def maximumHp: Int = _maximumHp

  override def opcode(): SendOpcode = SendOpcode.DAMAGE_MONSTER
}

File: engine/engine-model/src/main/java/tools/packet/monster/DamageMonsterFriendly.java
package tools.packet.monster;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record DamageMonsterFriendly(Integer objectId, Integer damage, Integer remainingHp,
                                    Integer maximumHp) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.DAMAGE_MONSTER;
   }
}